manifests/snmpv3_user.pp

```diff
@@ -64,14 +64,44 @@
   }
 
   if $create {
-    exec { "create-snmpv3-user-${title}":
-      path    => '/bin:/sbin:/usr/bin:/usr/sbin',
-      # TODO: Add "rwuser ${title}" (or rouser) to /etc/snmp/${daemon}.conf
-      command => "service ${service_name} stop ; sleep 5 ; \
-echo \"${cmd}\" >>${snmp::params::var_net_snmp}/${daemon}.conf",
-      user    => 'root',
-      require => [ Package['snmpd'], File['var-net-snmp'], ],
-      before  => Service[$service_name],
+    unless defined(Exec["stop-${service_name}"]) {
+      #
+      # TODO: update $command for different operating systems/releases
+      #
+      $command = "service ${service_name} stop ; sleep 5"
+
+      exec { "stop-${service_name}":
```

Comment:
Why is this even an exec resource? Can't we refresh the service?

Reply:
I'm looking to make some changes to this module to handle v3 engine IDs and other things.  I've noticed that /var/lib/net-snmp is being used for the configuration.  Why not use /etc/snmp instead?  The files under /var/lib/net-snmp say to not edit manually and would make the code for this a lot cleaner.

Reply:
> I'm looking to make some changes to this module to handle v3 engine IDs and other things. I've noticed that /var/lib/net-snmp is being used for the configuration. Why not use /etc/snmp instead? The files under /var/lib/net-snmp say to not edit manually and would make the code for this a lot cleaner.

@hanej  smoeding explains well. If you need to ready again, for instance have a look to:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html-single/system_administrators_guide/#sect-System_Monitoring_Tools-Net-SNMP-Configuring

Reply:
Even that documentation says to edit the config files under /etc/snmp.  I’ll create a PR with my changes.

Reply:
See the official manpage at http://www.net-snmp.org/docs/man/snmpd.examples.html

> Remember that these createUser directives should be defined in the /var/net-snmp/snmpd.conf file, rather than the usual location.